src/fhirtypes/ElementDefinition.ts

```diff
@@ -519,7 +498,11 @@ export class ElementDefinition {
    */
   private initializeElementType(rule: AddElementRule, fisher: Fishable): ElementDefinitionType[] {
     if (rule.types.length > 1 && !rule.path.endsWith('[x]')) {
-      throw new InvalidChoiceTypeRulePathError(rule);
+      const referenceTypes = rule.types.filter(t => t.isReference);
+      // Reference data type with multiple targets is not considered a choice data type.
+      if (referenceTypes.length < rule.types.length) {
```

Comment:
I think this could be simplified as:
```
if (rule.types.some(t => ! t.isReference)) {
```
or
```
if (! rule.types.every(t => t.isReference)) {
```

Reply:
But that could be just my opinion.  If you feel that the approach you took is easier to understand then feel free to ignore.

Reply:
good suggestion - fixed